src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -46,6 +46,22 @@ internal sealed class VariableState
             }
         }
 
+        /// <summary>
+        /// Represents the result of visiting an expression.
+        /// Contains a result type which tells us whether the expression may be null,
+        /// and an l-value type which tells us whether we can assign null to the expression.
```

Comment:
Should we track a pair of { Type, Slot } instead, where VisitRvalue would re-interpret the pair to an r-value result?

Tracking { Type, Slot } might also allow removing MakeSlot(BoundExpression) since the slot would already be calculated.

Reply:
I expect the `VisitResult` type will evolve. I think we'll use it to store multiple slots in conditional L-values. But that is beyond the present PR.

---
In reply to: [258182079](https://github.com/dotnet/roslyn/pull/33447#discussion_r258182079) [](ancestors = 258182079)

Reply:
From discussion with Chuck, we need to think about this alternative design some more. Will keep that separate from present PR.

---
In reply to: [258237980](https://github.com/dotnet/roslyn/pull/33447#discussion_r258237980) [](ancestors = 258237980,258182079)